src/algebra/big_operators/basic.lean

```diff
@@ -259,6 +259,18 @@ by rw [finset.prod, ← multiset.coe_prod, ← multiset.coe_map, finset.coe_to_l
 
 end to_list
 
+@[to_additive]
+lemma _root_.equiv.perm.prod_comp (σ : equiv.perm α) (s : finset α) (f : α → β)
+  (hs : {a | σ a ≠ a} ⊆ s) :
+  (∏ x in s, f (σ x)) = ∏ x in s, f x :=
+by { convert (prod_map _ σ.to_embedding _).symm, exact (map_perm hs).symm }
```

Comment:
The proof in terms of `prod_bij` looks something like
```suggestion
@[to_additive]
lemma _root_.equiv.perm.prod_comp (σ : equiv.perm α) (s : finset α) (f : α → β)
  (hs : {a | σ a ≠ a} ⊆ s) :
  (∏ x in s, f (σ x)) = ∏ x in s, f x :=
prod_bij'
  (λ x _, σ x) (λ x hx, begin
    sorry,
  end)
  (λ _ _, rfl)
  (λ x _, σ.symm x) (λ x hx, begin
    sorry,
  end)
  (λ x _, σ.symm_apply_apply _) (λ x _, σ.apply_symm_apply _)
```
Obviously this is much more verbose, but I also wonder whether there's a stronger lemma hiding here with a less weird condition than `{a | σ a ≠ a} ⊆ s`

Reply:
I can't think of any.